### DISCUSSION TITLE
Type ...UseFormReturn<ShareFormInputs, any, undefined> is not assignable to type React.ReactNode | React.ReactNode[]

### DISCUSSION THREAD

**Discussion by ricjohn-acosta** [shadcn-ui/ui#4914] [General]
I'm following the [form documentation](https://ui.shadcn.com/docs/components/textarea#form) for a `Textarea` input but I'm running into this type error when trying to use the spread operator.

![image](https://github.com/user-attachments/assets/b5c4bd1d-a710-4ea2-87a1-3e640c4fd4ff)

`form` here just comes from react-hook-library

e.g:

```
  const form = useForm({
    mode: "onChange",
    defaultValues: {...},
  });
```

Any ideas?

**Comment by wbsdickson** [comment]
same error with typescript 5.6.3
![image](https://github.com/user-attachments/assets/7bdd6e9a-63aa-4b67-92f6-a4f701a7a3b0)

Resolved the problem with downloading the typescript to 5.3.3

![image](https://github.com/user-attachments/assets/f9923a39-9892-4033-833e-1f5b001db862)

**Reply by illustudio** [reply]
@wbsdickson This was very helpful. Thanks to you, I was able to solve it. I was suffering with this problem for almost a week.

**Reply by gpradas-bjumper** [reply]
Same issue happened to me while kickstarting a react + vite project in march 2025. This is the only thing that worked. In my case Typescript was also complaining about typings inside Form sub-components, in the render attribute, with the field argument. 

Thanks!
